### PR TITLE
feat: added query params injection to api plugin url & headers

### PR DESCRIPTION
**Pull request by chesterkmr**
## Release Notes

- **New Features**
	- Enhanced API plugin functionality with the inclusion of global query parameters in requests.
  
- **Bug Fixes**
	- Updated dependencies across various packages to ensure compatibility and stability.
  
- **Documentation**
	- Changelogs updated to reflect new version entries and dependency updates for multiple projects.
  
- **Refactor**
	- Improved organization of imports and plugin management within the workflow runner and related plugins.
  
- **Tests**
	- New test suite added for helper functions, ensuring robust validation and error handling.

### CODE REVIEW

File: packages/workflow-core/src/lib/plugins/external-plugin/api-plugin/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+export const isBrowser = () => typeof window !== 'undefined';
+
+export const getGlobalQueryParams = () => {
+  if (!isBrowser()) return {};
+
+  const searchParams = new URLSearchParams(window.location.search);
+  const paramsObject: Record<string, string> = {};
+
+  searchParams.forEach((value, key) => {
+    paramsObject[key] = value;
```
Comment (tomer-shvadron) [inline packages/workflow-core/src/lib/plugins/external-plugin/api-plugin/helpers.ts]: Let's sanitize `value` to make sure it's not a security vulnerability

File: packages/workflow-core/src/lib/plugins/external-plugin/api-plugin/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  });
+
+  return paramsObject;
+};
```
[the remark this answers is not in the document]
Comment (tomer-shvadron) [reply packages/workflow-core/src/lib/plugins/external-plugin/api-plugin/helpers.ts]: @chesterkmr good idea ☝️

File: packages/workflow-core/src/lib/plugins/external-plugin/api-plugin/api-plugin.ts [hunk 3]
```diff
@@ -251,7 +255,12 @@ export class ApiPlugin {
     return { [returnArgKey]: isValid, errorMessage };
   }
 
-  async composeRequestHeaders(headers: HeadersInit, context: TContext) {
+  async composeRequestHeaders(headers: HeadersInit, _context: TContext) {
+    const context = {
```
Comment (tomer-shvadron) [inline packages/workflow-core/src/lib/plugins/external-plugin/api-plugin/api-plugin.ts:259]: Why do we need to add the query params to the context?

Comment (chesterkmr) [reply packages/workflow-core/src/lib/plugins/external-plugin/api-plugin/api-plugin.ts:259]: Method uses context as data source to replace tags. Just updating single data source here so query values also could be included. No mutations so should not affect `context` outside of this method call.

**Review by tomer-shvadron — APPROVED** [review]

**Pull request closed without merging**